tests/test_history.py

```diff
@@ -1329,6 +1329,41 @@ def handle_data(context, data):
             ' HistorySpec',
         )
 
+    def test_history_in_initialize(self):
+        algo_text = dedent(
+            """\
+            from zipline.api import history
+
+            def initialize(context):
+                history(10, '1d', 'price')
+
+            def handle_data(context, data):
+                pass
+            """
+        )
+
+        start = pd.Timestamp('2007-04-05', tz='UTC')
+        end = pd.Timestamp('2007-04-10', tz='UTC')
+
+        sim_params = SimulationParameters(
+            period_start=start,
+            period_end=end,
+            capital_base=float("1.0e5"),
+            data_frequency='minute',
+            emission_rate='daily',
+            env=self.env,
+        )
+
+        test_algo = TradingAlgorithm(
+            script=algo_text,
+            data_frequency='minute',
+            sim_params=sim_params,
+            env=self.env,
+        )
+
+        with self.assertRaises(HistoryInInitialize):
+            test_algo.initialize()
```

Comment:
In addition to calling `initialize()` directly we could also check that `HistoryInInitialize ` is raised when we call `.run()`. For example, we could add the following before hand: `source = RandomWalkSource(start=start, end=end)` and add the check:
```python
with self.assertRaises(HistoryInInitialize):
    test_algo.run(source)
```
Ultimately this will call `.initialize()`, but it would improve coverage. If you think it's unnecessary then just leave as is.